src/ports/postgres/modules/deep_learning/madlib_keras_wrapper.py_in

```diff
@@ -56,8 +58,13 @@ def clear_keras_session():
 def compile_and_set_weights(segment_model, compile_params, device_name,
                             previous_state, model_shapes):
     with K.tf.device(device_name):
-        compile_params = convert_string_of_args_to_dict(compile_params)
-        segment_model.compile(**compile_params)
+
+        optimizers = get_optimizers()
+        (opt_name,final_args,compile_dict) = parse_compile_params(compile_params)
+
+        segment_model.compile(optimizer=optimizers[opt_name](**final_args),
```

Comment:
What kind of error will it throw if `opt_name` is not a valid key in optimizers? Do we want to catch it by ourselves, or just throw whatever keras throws?